homeassistant/components/wyoming/__init__.py

```diff
@@ -25,23 +34,70 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if service is None:
         raise ConfigEntryNotReady("Unable to connect")
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = service
+    item = DomainDataItem(service=service)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = item
 
-    await hass.config_entries.async_forward_entry_setups(
-        entry,
-        service.platforms,
-    )
+    await hass.config_entries.async_forward_entry_setups(entry, service.platforms)
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    if (satellite_info := service.info.satellite) is not None:
+        # Create satellite device, etc.
+        item.satellite = _make_satellite(hass, entry, service)
+
+        # Set up satellite sensors, switches, etc.
+        await hass.config_entries.async_forward_entry_setups(entry, SATELLITE_PLATFORMS)
+
+        # Start satellite communication
+        entry.async_create_background_task(
+            hass,
+            item.satellite.run(),
+            f"Satellite {satellite_info.name}",
+        )
+
+        entry.async_on_unload(item.satellite.stop)
 
     return True
 
 
+def _make_satellite(
+    hass: HomeAssistant, config_entry: ConfigEntry, service: WyomingService
+) -> WyomingSatellite:
+    """Create Wyoming satellite/device from config entry and Wyoming service."""
+    satellite_info = service.info.satellite
+    assert satellite_info is not None
+
+    dev_reg = dr.async_get(hass)
+
+    # Use config entry id since only one satellite per entry is supported
+    satellite_id = config_entry.entry_id
+
+    device = dev_reg.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        identifiers={(DOMAIN, satellite_id)},
+        name=satellite_info.name,
+        suggested_area=satellite_info.area,
+    )
+
+    satellite_device = SatelliteDevice(
+        satellite_id=satellite_id,
+        device_id=device.id,
+    )
+
+    return WyomingSatellite(hass, service, satellite_device)
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry):
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Wyoming."""
-    service: WyomingService = hass.data[DOMAIN][entry.entry_id]
+    item: DomainDataItem = hass.data[DOMAIN][entry.entry_id]
 
     unload_ok = await hass.config_entries.async_unload_platforms(
         entry,
-        service.platforms,
+        item.service.platforms,
```

Comment:
do these include the `SATELLITE_PLATFORMS` ?

Reply:
It did not! Added.